src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -903,36 +870,29 @@ def encryption_attachall(self, job):
                         self.logger.warn('Unable to attach GELI provider: %s', msg)
                         failed_drive += 1
 
-                job = self.middleware.call_sync('zfs.pool.import', pool['guid'], {
-                    'altroot': '/mnt',
-                })
-                job.wait_sync()
-                if job.error:
+                try:
+                    self.middleware.call_sync('zfs.pool.import_pool', pool['guid'], {
+                        'altroot': '/mnt',
+                    })
+                except Exception as e:
                     failed_volume += 1
+                    self.logger.error('Failed to import %s pool: %s', pool['name'], str(e))
 
-            if failed_drive > 0:
-                job.set_progress(None, f'{failed_drive} can not be attached.')
-                self.logger.error('%d can not be attached.', failed_drive)
+        if failed_drive > 0:
+            job.set_progress(None, f'{failed_drive} can not be attached.')
```

Comment:
`{failed_drive}` is an integer which doesn't represent the actual geli provider. The `encrypted_disk['encrypted_provider']` should be what you want.

Reply:
I think original author meant something like `2 drives cannot be attached` as not sure if in progress it would be of much benefit telling the paths and they can be a lot too if we have more disks in the pool.
Nice catch ; ) I'll make amends, it will look really weird right now : p